### ISSUE TITLE
Duplicate (ctrl-D) should not work while editing a script

### ISSUE THREAD

**Issue by paryl** [godotengine/godot#16126]
I searched for this and couldn't find it. Sorry if it's a duplicate.

Godot 3 rc2
Windows 10 Pro

Select a scene object, then start editing a script. If you hit ctrl-D instead of ctrl-S to save it, you'll duplicate the last object selected. Given that D and S are next to each other, this happens an embarrassing amount to me, requiring that I then delete duplicate objects.

**Comment by bfloch** [comment]
I can make a pull request if handling it from the SceneTreeDock satisfies it:

```C++
 diff --git a/editor/scene_tree_dock.cpp b/editor/scene_tree_dock.cpp                           
index 41ee5ab28..902018ab4 100644              
--- a/editor/scene_tree_dock.cpp               
+++ b/editor/scene_tree_dock.cpp               
@@ -421,6 +421,11 @@ void SceneTreeDock::_tool_selected(int p_tool, bool p_confirm_override) { 
                        if (!edited_scene)     
                                break;         
                                               
+                       EditorPlugin *editor_plugin = EditorNode::get_singleton()->get_editor_plugin_screen();                                                                                 
+                       Control *focus_owner = get_focus_owner();                              
+                       if (editor_plugin->get_name() == "Script" && focus_owner && focus_owner->get_class() == "TextEdit")                                                                    
+                               break;         
+                                              
                        if (editor_selection->is_selected(edited_scene)) {                     
                                               
                                current_option = -1;                                            
```
This still allows for duplication if the scene tree is selected specifically in script editor mode. 

Maybe someone knows a way to handle this in a more generically?

**Comment by Neicul** [comment]
Also probably make Ctrl+D duplicate the current active line (if there is no selection) or duplicate the selected text (like copy + paste)

**Comment by akien-mga** [comment]
Fixed by #16373.
